python/ray/_raylet.pyx

```diff
@@ -3652,21 +3652,46 @@ cdef class CoreWorker:
             check_status(CCoreWorkerProcess.GetCoreWorker()
                          .ExperimentalChannelSetError(c_object_id))
 
-    def experimental_channel_register_writer(self, ObjectRef object_ref):
+    def experimental_channel_register_writer_local(self, ObjectRef object_ref):
```

Comment:
Ideally we should try to minimize the interfaces for the language frontends and the C++ CoreWorker. Do you think we could have a single experimental_channel_register_writer function that handles the registration for both local and remote readers? And same for registering readers? Then we can implement the logic within ExperimentalMutableObjectProvider. It would be easier to unit-test the code this way.